iwyu_path_util.cc

```diff
@@ -190,7 +193,16 @@ string ConvertToQuotedInclude(const string& filepath) {
   for (Each<HeaderSearchPath> it(&search_paths); !it.AtEnd(); ++it) {
     // All header search paths have a trailing "/", so we'll get a perfect
     // quoted include by just stripping the prefix.
-    if (StripLeft(&path, it->path)) {
+    bool matched = StripLeft(&path, it->path);
+    if (!matched && !IsAbsolutePath(it->path)) {
```

Comment:
The nested conditionals here are killing me... Do you think it's possible to simplify this if you switch to guard-clause-style ifs with early returns?